### DISCUSSION TITLE
Add the ability to exclude pages from the @astrojs/sitemap within the getStaticPaths function.

### DISCUSSION THREAD

**Discussion by Reyno-Skeps** [withastro/roadmap#893] [Proposal]
### Body

# Summary

Add the ability to exclude pages from the [@astrojs/sitemap](https://docs.astro.build/en/guides/integrations-guide/sitemap/) within the `getStaticPaths` function.

# Background & Motivation

Currently we have the ability to filter out certain urls via the [filter](https://docs.astro.build/en/guides/integrations-guide/sitemap/#filter) or [serialize](https://docs.astro.build/en/guides/integrations-guide/sitemap/#serialize) property in the config. This unfortunately doesn't work for websites connected via a CMS since these values are hardcoded.

In our CMS (Storyblok) users can set the robots property for a page (index or noindex) to in- or exclude them from the search results. This works perfectly for the meta tag but the URL is still included in the sitemap giving conflicting results to search engines.

# Goals

- Make it possible to exclude pages dynamically from the sitemap
- Don't use hardcoded values to filter pages.

# Example

One solution would be to set a prop within the `getStaticPaths` function. This will make sure the page can still be statically generated without adding it to the sitemap.

```js
export const getStaticPaths = async () => {
  const pages = await fetch( '/cms/pages' );

  return pages.map( ( page ) => ( {
    params: { slug: page.slug },
    props: {
      excludeSitemap: true, // page.robots === 'noindex'
    }
  } ) );
};
```

Or as separate property:

```js
export const getStaticPaths = async () => {
  const pages = await fetch( '/cms/pages' );

  return pages.map( ( page ) => ( {
    params: { slug: page.slug },
    sitemap: {
      exclude: true, // page.robots === 'noindex'
    }
  } ) );
};
```

**Comment by florian-lefebvre** [comment]
You may be interested in https://inox-tools.vercel.app/sitemap-ext by @Fryuni

**Reply by Reyno-Skeps** [reply]
Yes thank you, this is exactly what I was looking for. Love to see it as a core feature someday but this works perfectly for now.
